app/services/fastboot.js

```diff
@@ -57,6 +58,15 @@ const Shoebox = Ember.Object.extend({
     this.set(key, shoeboxItem);
 
     return shoeboxItem;
+  },
+
+  has(key) {
+    if (this.get('fastboot.isFastBoot')) {
+      let shoebox = this.get('fastboot._fastbootInfo.shoebox');
+      return shoebox && shoebox.hasOwnProperty(key);
+    } else {
+      return !!document.querySelector(`#shoebox-${key}`);
```

Comment:
if shoebox is for random data (that we likely don't want displayed) should we use meta or script tags?